tests/unit-tests/src/VuFindTest/OaiPmh/HarvesterTest.php

```diff
@@ -399,17 +386,54 @@ public function testArbitraryOaiError()
 
         $comm = $this->getMockCommunicator();
         $expectedSettings = ['metadataPrefix' => 'oai_dc'];
-        $comm->expects($this->once())->method('request')
-            ->with($this->equalTo('ListRecords'), $this->equalTo($expectedSettings))
-            ->will(
-                $this->returnValue(
-                    simplexml_load_string($this->getArbitraryErrorResponse())
-                )
+        $comm->expects($this->exactly(2))->method('request')
+            ->withConsecutive(
+                ['Identify', []],
+                ['ListRecords', $expectedSettings],
+            )
+            ->willReturnOnConsecutiveCalls(
+                simplexml_load_string($this->getFakeIdentifyResponse()),
+                simplexml_load_string($this->getArbitraryErrorResponse())
             );
         $harvester = $this->getHarvester(
             ['dateGranularity' => 'YYYY-MM-DDThh:mm:ssZ'],
             $comm
         );
         $harvester->launch();
     }
+
+    /**
+     * With a single ListRecords call test that the persisted harvest end date
+     * is formatted appropriately when day granularity is detected.
+     *
+     * @return void
+     */
+    public function testSimpleListRecordsGranularityHandling()
```

Comment:
This method is my attempt to retain some coverage for granularity detection. This simply ensures that we get a YYYY-MM-DD formatted end date when date granularity is declared.